core/arch/aarchxx/mangle.c

```diff
@@ -219,6 +219,86 @@ insert_clear_eflags(dcontext_t *dcontext, clean_call_info_t *cci,
      */
 }
 
+#ifdef AARCH64
+/* Use first register to determine if we are dealing with GPRs or SIMD registers */
+#define IF_GPR_ELSE(first_reg, then, else_)  (first_reg == DR_REG_X0 ? then : else_)
```

Comment:
It is kind of strange to me, usually I would use something like
```
# define GET_FIRST_REG(is_gpr)   (is_gpr) ? DR_REG_X0 : DR_REG_Q0
# define GET_REG_OFFSET(is_gpr, reg) (is_gpr) ? REG_OFFSET(DR_REG_X0 +  reg) : reg1*sizeof(dr_simd_t).
```

IF_GPR_ELSE first gets is_gpr based on the reg, and then pick value. It is one macro can be used in several places, but every time I see it, I have to think twice about what it does.
IMHO, GET_... is more straight-forward and easier to understand/read.
Also, passing is_reg or is_simd seems better than first_reg, esp you will use reg_id_t for first_reg, and uint for reg1, kind of confusing.